aten/src/ATen/Dispatch.h

```diff
@@ -143,6 +143,7 @@
       AT_PRIVATE_CASE_TYPE(at::ScalarType::Long, int64_t, __VA_ARGS__)       \
       AT_PRIVATE_CASE_TYPE(at::ScalarType::Short, int16_t, __VA_ARGS__)      \
       AT_PRIVATE_CASE_TYPE(at::ScalarType::Half, at::Half, __VA_ARGS__)      \
+      AT_PRIVATE_CASE_TYPE(at::ScalarType::Bool, bool, __VA_ARGS__)          \
```

Comment:
I don't think this solution is scalable; it's totally not obvious what the state of bool is wrt to these macros.  The issues seems to be with Half and Bool.

Can you figure out something better here?

Reply:
Could you elaborate, please? I'm not sure if i follow what is wrong here